src/remote.cc

```diff
@@ -832,18 +832,17 @@ Server::Server(String session_name)
     m_listener.reset(new FDWatcher{listen_sock, FdEvents::Read, accepter});
 }
 
-bool Server::rename_session(StringView name)
+bool Server::rename_session(StringView name, bool force)
 {
     if (not all_of(name, is_identifier))
         throw runtime_error{format("invalid session name: '{}'", name)};
 
     String old_socket_file = session_path(m_session);
     String new_socket_file = session_path(name);
 
-    if (file_exists(new_socket_file))
-        return false;
-
-    if (rename(old_socket_file.c_str(), new_socket_file.c_str()) != 0)
+    if ((file_exists(new_socket_file) or
+         rename(old_socket_file.c_str(), new_socket_file.c_str()) != 0)
+         and not force)
```

Comment:
So, if the file exist and we are forcing, we wont rename the scoket file, and but we will still believe that we did. That means we will remove the wrong socket (the one from another session) in close_session, and that any use of $kak_session will refer to another server, that does not look like desired behaviour.